File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function permutations(items, set) {
    if(set === 1) {
        return items.map((item) => [item]);
    }
    
    let results = [];
    items.forEach((item, index) => {
        let left = [...items.slice(0, index), ...items.slice(index + 1)];
        let permutation = permutations(left, set - 1);
        permutation.forEach((combi) => results.push([item, ...combi]));
    })
    
    return results;
}


function convert_to_number(string_numbers) {
    return Number(string_numbers.join(""));
}


function is_prime_number(number) {
    if(number < 2) {
        return false;
    }
    
    for(let i = 2; i <= Math.sqrt(number); i++) {
        if(!(number % i)) {
            return false;
        }
    }
    
    return true;
}


function solution(numbers) {
    let answer = 0;
    let length = numbers.length;
    let entire_combinations = {};
    
    numbers = numbers.split("");
    
    for(let i = 1; i <= length; i++) {
        for(let permunation of permutations(numbers, i)) {
            let number = convert_to_number(permunation);
            
            if(!(!!entire_combinations[number])) {
                entire_combinations[number] = true;
                if(is_prime_number(number)) {
                    answer++;
                }
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
from itertools import permutations


def perm2num(perm):
    nums = []
    for p in perm:
        nums.append(str(p))
    return int("".join(nums))
    

def is_prime(num):
    if num < 2:
        return False
    for i in range(2, round(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    return True
    
    
def solution(numbers):
    answer = 0
    already = {}
    numbers = list(map(int, list(numbers)))
    
    for num in range(1, len(numbers) + 1):
        for perm in permutations(numbers, num):
            each_num = perm2num(perm)
            if each_num not in already:
                already[each_num] = True
                if is_prime(each_num):
                    answer += 1
    
    return answer